### PR TITLE
make simple samples runnable

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description

* Update simple config examples so you can copy paste and run with no editing
* Clearly link to dev hub and workflows doc for further reading
* Put VS Code extension paragraphs in their own section to make doc easier to follow
* Also change the name of the configuration reference to "configuration reference" .... currently it's hard to find, and searching doesn't bring it up

# Reasons
I want to link to this page as example of basic configs to get people started, it would be much easier for people if they can just copy and paste the snippets, and they can easily find the dev hub to update the docker images if they want

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/sample-config.md [hunk 1]
```diff
@@ -316,7 +279,10 @@ workflows:
 ### Approval job
 {: #approval-job }
 
-The example below shows a sequential job workflow with an approval step. The `build` job runs, then the `test` job, then a `hold` job, with `type: approval` ensures the workflow waits for manual approval before the `deploy` job can run. Refer to the [Workflows]({{ site.baseurl }}/workflows) document for complete details about orchestrating job runs with concurrent, sequential, and manual approval workflows.
+The example below shows a sequential workflow with an [approval step](/docs/configuration-reference/#type). The `build` job runs, then the `test` job, then a `hold` job, with `type: approval` ensures the workflow waits for manual approval in the CircleCI web app before the `deploy` job can run. All jobs run in Docker containers using the base image provided by CircleCI.
```
Comment (kbatuigas) [inline jekyll/_cci2/sample-config.md:282]:
```suggestion
The example below shows a sequential workflow with an [approval step](/docs/configuration-reference/#type). The `build` job runs, then the `test` job, then a `hold` job, with `type: approval` ensuring the workflow waits for manual approval in the CircleCI web app before the `deploy` job can run. All jobs run in Docker containers using the base image provided by CircleCI.
```
Wasn't sure about the phrasing - something like this sounds more understandable to me.

**Review by kbatuigas — CHANGES_REQUESTED** [review]
Just a couple of nits and clarifications, but I think it looks good

File: jekyll/_cci2/sample-config.md [hunk 2]
```diff
@@ -27,9 +30,13 @@ The CircleCI VS Code extension is available to download on the [VS Code marketpl
 ### Concurrent workflow
 {: #concurrent-workflow }
 
-The configuration example below shows a concurrent  workflow in which the `build` and `test` jobs run at the same time. Refer to the [Workflows]({{ site.baseurl }}/workflows) document for complete details about orchestrating job runs with concurrent, sequential, and manual approval workflows.
+The configuration example below shows a concurrent  workflow in which the `build` and `test` jobs run at the same time. Both jobs are run in Docker containers using the base image provided by CircleCI.
+
+* Refer to the [Workflows](/docs/workflows) document for complete details about orchestrating job runs with concurrent, sequential, and manual approval workflows.
+* Refer to the [developer hub convenience images](https://circleci.com/developer/images?imageType=docker) page to find out about available Docker images for running your jobs.
```
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/sample-config.md:36]: Good question

File: jekyll/_cci2/sample-config.md [hunk 3]
```diff
@@ -27,9 +30,13 @@ The CircleCI VS Code extension is available to download on the [VS Code marketpl
 ### Concurrent workflow
 {: #concurrent-workflow }
 
-The configuration example below shows a concurrent  workflow in which the `build` and `test` jobs run at the same time. Refer to the [Workflows]({{ site.baseurl }}/workflows) document for complete details about orchestrating job runs with concurrent, sequential, and manual approval workflows.
+The configuration example below shows a concurrent workflow in which the `build` and `test` jobs run at the same time. Both jobs are run in Docker containers using the base image provided by CircleCI.
+
+* Refer to the [Workflows](/docs/workflows) document for complete details about orchestrating job runs with concurrent, sequential, and manual approval workflows.
+* Refer to the [developer hub convenience images](https://circleci.com/developer/images?imageType=docker) page to find out about available Docker images for running your jobs.
```
Comment (rosieyohannan) [inline jekyll/_cci2/sample-config.md:36]:
```suggestion
* Refer to the [Developer Hub convenience images](https://circleci.com/developer/images?imageType=docker) page to find out about available Docker images for running your jobs.
```

**Pull request merged**